pkg/metrics/metrics.go

```diff
@@ -0,0 +1,234 @@
+// errors (except for AlreadyRegisteredError) during registration of the metric.
+func InitGauge(r prometheus.Registerer, opts prometheus.GaugeOpts) prometheus.Gauge {
+	g := prometheus.NewGauge(opts)
+	rg := registerGauge(r, g)
+	return rg
+}
+
+// InitHistogram initializes a new histogram metric. The method panics if there are any
+// errors (except for AlreadyRegisteredError) during registration of the metric.
+func InitHistogram(r prometheus.Registerer, opts prometheus.HistogramOpts) prometheus.Histogram {
+	h := prometheus.NewHistogram(opts)
+	rh := registerHistogram(r, h)
+	return rh
+}
+
+// registerCounterVec registers the CounterVec with the provided Registerer. It panics if the
+// type check fails
+func registerCounterVec(r prometheus.Registerer, g *prometheus.CounterVec) *prometheus.CounterVec {
+	c := registerMetricOrDie(r, g)
+	gv, ok := c.(*prometheus.CounterVec)
+	if !ok {
+		panic("failed type check for CounterVec")
+	}
+	return gv
+}
+
+// registerHistogramVec registers the HistogramVec with the provided Registerer. It panics if the
+// type check fails
+func registerHistogramVec(r prometheus.Registerer, h *prometheus.HistogramVec) *prometheus.HistogramVec {
+	c := registerMetricOrDie(r, h)
+	v, ok := c.(*prometheus.HistogramVec)
+	if !ok {
+		panic("failed type check for HistogramVec")
+	}
+	return v
+}
+
+// registerGaugeVec registers the GaugeVec with the provided Registerer. It panics if the
+// type check fails.
+func registerGaugeVec(r prometheus.Registerer, g *prometheus.GaugeVec) *prometheus.GaugeVec {
+	c := registerMetricOrDie(r, g)
+	gv, ok := c.(*prometheus.GaugeVec)
+	if !ok {
+		panic("failed type check for GaugeVec")
+	}
+	return gv
+}
+
+// registerCounter registers the Counter with the provided Registerer. It panics if the
+// type check fails
+func registerCounter(r prometheus.Registerer, ctr prometheus.Counter) prometheus.Counter {
+	c := registerMetricOrDie(r, ctr)
+	rg, ok := c.(prometheus.Counter)
+	if !ok {
+		panic("failed type check for Counter")
+	}
+	return rg
+}
+
+// registerHistogram registers the Histogram with the provided Registerer. It panics if the
+// type check fails
+func registerHistogram(r prometheus.Registerer, g prometheus.Histogram) prometheus.Histogram {
+	c := registerMetricOrDie(r, g)
+	rg, ok := c.(prometheus.Histogram)
+	if !ok {
+		panic("failed type check for Histogram")
+	}
+	return rg
+}
+
+// registerGauge registers the Gauge with the provided Registerer. It panics if the
+// type check fails
+func registerGauge(r prometheus.Registerer, g prometheus.Gauge) prometheus.Gauge {
+	c := registerMetricOrDie(r, g)
+	rg, ok := c.(prometheus.Gauge)
+	if !ok {
+		panic("failed type check for Gauge")
+	}
+	return rg
+}
+
+// registerMetricOrDie is a helper to register a metric and log registration errors. If the metric
+// already exists, then it will be logged and the metric is returned. For other errors, the method
+// panics.
```

Comment:
```suggestion
// registerMetricOrDie is a helper to register a metric and log registration
// errors. If the metric already exists, then it will be logged, and the metric
// is returned. For other errors, the method panics.
```

Reply:
Done